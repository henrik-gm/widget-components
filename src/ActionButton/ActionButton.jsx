import React from 'react';
import PropTypes from 'prop-types';
import styles from './ActionButton.scss';

/* eslint-disable react/prefer-stateless-function */
class ActionButton extends React.Component {
   render() {
      let buttonClassName = `${styles.button} `;
      let buttonTextClassName = `${styles.buttonText} `;

      if (this.props.type === 'primary') {
         buttonClassName += 'KambiWidget-primary-color';
         buttonTextClassName += 'KambiWidget-primary-color';
      } else {
         buttonClassName += 'KambiWidget-card-text-color';
         buttonTextClassName += 'KambiWidget-card-text-color';
      }

      return (
         <div className={styles.container}>
            <button
               disabled={this.props.disabled}
               className={buttonClassName}
               onClick={this.props.action}
            />
            <div
               className={buttonTextClassName}
               style={this.props.disabled ? { opacity: 0.4 } : { opacity: 1 }}
            >
               { this.props.children }
            </div>
         </div>
      );
   }
}

ActionButton.propTypes = {
   action: PropTypes.func.isRequired,
   children: PropTypes.node.isRequired,
   disabled: PropTypes.bool,
   type: PropTypes.oneOf(['primary', 'secondary'])
};

ActionButton.defaultProps = {
   disabled: false,
   type: 'primary'
};

export default ActionButton;
