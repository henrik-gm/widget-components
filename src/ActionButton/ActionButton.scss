:local(.container) {
   position: relative;
   display: flex;
   justify-content: center;
   align-items: center;
   height: 36px;
   padding: 0 10px;
}

:local(.button) {
   position: absolute;
   top: 0;
   left: 0;
   height: 100%;
   width: 100%;
   overflow: hidden;
   padding: 0;
   margin: 0;
   opacity: 0;
   border: none;
   background-color: currentColor;
   cursor: pointer;
   transition: background-color 0.2s ease-out, color 0.2s ease-out;
   &:hover {
      opacity: 0.12;
   }
   &:active {
      opacity: 0.4;
   }
   &[disabled]:hover, &[disabled]:active {
      opacity: 0;
      cursor: not-allowed;
   }
}

:local(.buttonText) {
   text-transform: uppercase;
   font-weight: bold;
   background-color: transparent;
   pointer-events: none;
   font-size: 14px;
   text-align: left;
}
