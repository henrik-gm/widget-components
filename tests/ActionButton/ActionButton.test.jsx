/* eslint-env jest */
import React, { Children } from 'react';
import ActionButton from '../../src/ActionButton/ActionButton';
import { shallow } from 'enzyme';
import ReactTestUtils from 'react-addons-test-utils';

const renderer = ReactTestUtils.createRenderer();

describe('ActionButton DOM rendering', () => {

   it('renders correctly with default props', () => {
      expect(renderer.render(
         <ActionButton action={() => {}}>Test</ActionButton>
      )).toMatchSnapshot();
   });

   it('renders correctly when disabled', () => {
      expect(renderer.render(
         <ActionButton
            action={() => {}}
            disabled={true}
         >
            Test
         </ActionButton>
      )).toMatchSnapshot();
   });

   it('renders correctly with type set to \'secondary\'', () => {
      expect(renderer.render(
         <ActionButton
            action={() => {}}
            type='secondary'
         >
            Test
         </ActionButton>
      )).toMatchSnapshot();
   });

});

describe('ActionButton behaviour', () => {

   it('handles clicks correctly', () => {
      const actionMock = jest.fn();

      const wrapper = shallow(
         <ActionButton action={actionMock}>Test</ActionButton>
      );

      expect(actionMock).not.toHaveBeenCalled();

      wrapper.find('button').simulate('click');

      expect(actionMock).toHaveBeenCalledTimes(1);
   });

});
