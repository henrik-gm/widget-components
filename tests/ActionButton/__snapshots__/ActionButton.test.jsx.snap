// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionButton DOM rendering renders correctly when disabled 1`] = `
<div
  className="container"
>
  <button
    className="button KambiWidget-primary-color"
    disabled={true}
    onClick={[Function]}
  />
  <div
    className="buttonText KambiWidget-primary-color"
    style={
      Object {
        "opacity": 0.4,
      }
    }
  >
    Test
  </div>
</div>
`;

exports[`ActionButton DOM rendering renders correctly with default props 1`] = `
<div
  className="container"
>
  <button
    className="button KambiWidget-primary-color"
    disabled={false}
    onClick={[Function]}
  />
  <div
    className="buttonText KambiWidget-primary-color"
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    Test
  </div>
</div>
`;

exports[`ActionButton DOM rendering renders correctly with type set to 'secondary' 1`] = `
<div
  className="container"
>
  <button
    className="button KambiWidget-card-text-color"
    disabled={false}
    onClick={[Function]}
  />
  <div
    className="buttonText KambiWidget-card-text-color"
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    Test
  </div>
</div>
`;
